### PR TITLE
feat: use dep group

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: lib/ckb/api.rb [hunk 1]
```diff
@@ -18,10 +19,12 @@ def initialize(host: CKB::RPC::DEFAULT_URL, mode: MODE::TESTNET)
       @rpc = CKB::RPC.new(host: host)
       if mode == MODE::TESTNET
         # Testnet system script code_hash
-        expected_code_hash = "0xa76801d09a0eabbfa545f1577084b6f3bafb0b6250e7f5c89efcfd4e3499fb55"
+        expected_code_hash = "0x664b03c55410dd7d694a73779fa86edd7f069ee17fd79b0fef4c4fec0377b9a6"
         # For testnet chain, we can assume the second cell of the first transaction
         # in the genesis block contains default lock script we can use here.
         system_cell_transaction = genesis_block.transactions.first
+
+        system_scipt_group_transaction = genesis_block.transactions[1]
```
Comment (ashchan) [inline lib/ckb/api.rb]:
```suggestion
        system_script_group_transaction = genesis_block.transactions[1]
```

File: lib/ckb/api.rb [hunk 2]
```diff
@@ -32,6 +35,11 @@ def initialize(host: CKB::RPC::DEFAULT_URL, mode: MODE::TESTNET)
         raise "System script code_hash error!" unless code_hash == expected_code_hash
 
         set_system_script_cell(out_point, code_hash)
+        group_out_point = Types::OutPoint.new(
+          tx_hash: system_scipt_group_transaction.hash,
```
Comment (ashchan) [inline lib/ckb/api.rb]:
```suggestion
          tx_hash: system_script_group_transaction.hash,
```

File: lib/ckb/api.rb [hunk 3]
```diff
@@ -51,6 +59,11 @@ def set_system_script_cell(out_point, code_hash)
       @system_script_code_hash = code_hash
     end
 
+    # @param out_point [CKB::Types::OutPoint]
+    def set_system_script_group_cell(out_point)
```
Comment (ashchan) [inline lib/ckb/api.rb]: The func name is no longer suitable. It's not setting a cell but an out point.

File: lib/ckb/api.rb (hunk 1 above)
Comment (shaojunda) [reply lib/ckb/api.rb]: thanks, fixed.

File: lib/ckb/api.rb (hunk 3 above)
Comment (shaojunda) [reply lib/ckb/api.rb]: this cell is not equal with the cell on CKB, but I will use another name. thanks.

File: lib/ckb/api.rb [hunk 4]
```diff
@@ -40,29 +46,22 @@ def initialize(host: CKB::RPC::DEFAULT_URL, mode: MODE::TESTNET)
...
 
     # @param out_point [CKB::Types::OutPoint]
-    # @param code_hash [String] "0x..."
-    def set_system_script_cell(out_point, code_hash)
-      @system_script_out_point = out_point
-      @system_script_code_hash = code_hash
+    # @param secp_cell_type_hash [String] 0x...
+    def set_secp_group_dep(out_point, secp_cell_type_hash)
+      @secp_group_out_point = out_point
+      @secp_cell_type_hash = secp_cell_type_hash
```
Comment (classicalliu) [inline lib/ckb/api.rb]: Why not set `secp_cell_code_hash` here.

File: lib/ckb/api.rb [hunk 5]
```diff
@@ -140,6 +139,10 @@ def compute_transaction_hash(transaction)
       rpc.compute_transaction_hash(transaction.to_h)
     end
 
+    def compute_script_hash(script_h)
```
Comment (classicalliu) [inline lib/ckb/api.rb]: Should using `Types::Script` here.

File: lib/ckb/wallet.rb [hunk 6]
```diff
@@ -27,10 +28,13 @@ def initialize(api, key, skip_data_and_type: true)
       @addr = Address.from_pubkey(@pubkey)
       @address = @addr.to_s
       @skip_data_and_type = skip_data_and_type
+      raise "Wrong hash_type, hash_type should be Data or Type" if !%w(data type).include?(hash_type)
```
Comment (classicalliu) [inline lib/ckb/wallet.rb]:
```suggestion
      raise "Wrong hash_type, hash_type should be `data` or `type`" unless %w(data type).include?(hash_type)
```

File: lib/ckb/wallet.rb [hunk 7]
```diff
@@ -27,10 +28,13 @@ def initialize(api, key, skip_data_and_type: true)
...
+
+      @hash_type = hash_type
     end
 
-    def self.from_hex(api, privkey)
-      new(api, Key.new(privkey))
+    def self.from_hex(api, privkey, hash_type = "type")
```
Comment (classicalliu) [inline lib/ckb/wallet.rb]:
```suggestion
    def self.from_hex(api, privkey, hash_type: "type")
```

File: lib/ckb/api.rb (hunk 4 above)
Comment (shaojunda) [reply lib/ckb/api.rb]: because I don't want put `secp_cell_code_hash` to `set_secp_group_dep`

Comment (classicalliu) [reply lib/ckb/api.rb]: ok

File: lib/ckb/api.rb (hunk 5 above)
Comment (shaojunda) [reply lib/ckb/api.rb]: this method called in the Script.

Comment (shaojunda) [reply lib/ckb/api.rb]: thanks, I will change that.

File: lib/ckb/wallet.rb (hunk 7 above)
Comment (shaojunda) [reply lib/ckb/wallet.rb]: thanks, fixed.

File: lib/ckb/wallet.rb (hunk 6 above)
Comment (shaojunda) [reply lib/ckb/wallet.rb]: thanks, fixed.

**Comment by shaojunda** [comment]
`system code hash` may change, so I hold this pr.

**Pull request merged**